services/github/src/api.js

```diff
@@ -330,6 +338,18 @@ builder.declare({
           return resolve(res, 400, 'Only rerequested for check runs is supported');
         }
 
+        if (body?.sender?.type?.toLowerCase() === 'bot') {
```

Comment:
this might be a questionable solution, but it's the only way to differentiate when someone clicks `Re-run` in the UI (thus human) vs it was triggered by us when we detected that task is being rerun through api, so we are forced to call `rerequestRun()` as a bot user